docs/api/adc_api_requests.rst

```diff
@@ -0,0 +1,625 @@
+
+Note that the ``not`` operator is different from a logical NOT
+operator, and the logical NOT is not needed as the other operators
+provide negation.
+
+The ``field`` operand specifies a fully qualified property name in the AIRR
+Data Model. Fully qualified AIRR properties are either a JSON/YAML base type (``string``, ``number``,
+``integer``, or ``boolean``) or an array of one of these base types (some AIRR fields are arrays
+e.g. ``study.keywords_study``).
+The Fields section below describes the available queryable fields.
+
+The ``value`` operand specifies one or more values when evaluating the
+operator for the ``field`` operand.
+
+*Queries Against Arrays*
+
+A number of fields in the AIRR Data Model are arrays, such as
+``study.keywords_study`` which is an array of strings or
+``subject.diagnosis`` which is an array of ``Diagnosis`` objects. A
+query operator on an array field will apply that operator to each
+entry in the array to decide if the query filter is satisfied. The
+behavior is different for various operators. For operators such as
+``=`` and ``in``, the filter behaves like the Boolean ``OR`` over the
+array entries, that is if **any** array entry evaluates to true then
+the query filter is satisfied. For operators such as ``!=`` and
+``exclude``, the filter behaves like the Boolean ``AND`` over the
+array entries, that is **all** array entries must evaluate to true for
+the query filter to be satisfied.
+
+For complex queries over arrays, it is necessary to compose complex queries into
+more than one query. For example consider the following subject:
+
+.. code-block:: bash
+
+  * Subject
+    * diagnosis
+      (Diagnosis record 1)
+        * disease_diagnosis: "rheumatoid arthritis"
+        * disease_length: "20 years"
+      (Diagnosis record 2)
+        * disease_diagnosis: "pancreatic ductal adenocarcinoma"
+        * disease_length: "6 months"
+
+
+If the end result that is required it to find all disease diagnoses of "pancreatic ductal adenocarcinoma"
+that have a disease length of over 10 years, searching for ``disease_diagnosis = pancreatic ductal adenocarcinom`` and ``disease_length > 10``
+will result in the above Subject being returned, even though the subject has not had pancreatic ductal adenocarcinom for more than 10 years.
+This is because there is a diagnosis of pancreatic ductal adenocarcinom and a disease length
+of more than 10 years but from a different diagnoses. This is a correct response to the query, but does not return the desired outcome.
+
+In order to acheive the desired outcome, it is necessary to search for one of the conditions (e.g. ``disease_diagnosis = pancreatic ductal adenocarcinom``),
+compile a list of ``repertoire_ids`` that meet that condition, and then search for the second condition (e.g. ``disease_length > 10``)
+across those ``repertoire_ids``. 
```

Comment:
@bcorrie I don't have strong opinions on the "global" vs. "local" handling of such requests, but I would like to note that the text here reads different from what we [discussed in the ticket](https://github.com/airr-community/airr-standards/issues/623#issuecomment-1214496963). I you agree, we should note this in the issue, too.

Reply:
@bussec the above documentation reflects this comment: https://github.com/airr-community/airr-standards/issues/623#issuecomment-1208073359

Or at least it was meant to. Not sure I am clear what you think is missing...

Reply:
@bcorrie Yes, it reflects your comment, which however pre-dates our discussions on this topics in [call 2022-08](https://github.com/airr-community/airr-standards/wiki/Minutes_Standards_2022-08), which is what I was referring to. My recollection of the discussion was that "local" behavior is what most users would expect, and that therefore it is the way how the API should behave. In addition, the potential problems resulting from such an implementation with searches requiring a "global" behavior, should be documented and potential work-arounds be provided. Again, we can reverse this decision, but then this needs to be done explicitly, we should not just drop it.

In addition, from a logical side, I have problems understanding why the procedure described in lines 346-348 should yield the desired outcome -- i.e., simulating "local" behavior, which would **not** return the example record -- as the first query would clearly return the repertoire containing this record, but so would the second query, even when filtering on those repertoires which were already returned by the first query.

Reply:
Fine with me, let's get this thing merged ;-)